[BUG] Analysis crashes on lazy initialisation to anonymous object

**Describe the bug**
A kotlin property which uses a lazy initialiser which contains an anonymous object

**To Reproduce**
Run arrow analysis against a file containing this property:
```kotlin
val p by lazy { object : Any() {} }
```

The following exception is thrown during compile:
```
org.jetbrains.kotlin.util.KotlinFrontEndException: Exception while analyzing expression at (3,17) in /home/wesley/work/projects/arrow-analysis-test/src/main/kotlin/ca/cutterslade/arrowanalysis/test/lazy-init.kt

Attachments:
expression.kt
object : Any() {}
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.logOrThrowException(ExpressionTypingVisitorDispatcher.java:253)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.lambda$getTypeInfo$0(ExpressionTypingVisitorDispatcher.java:224)
	at org.jetbrains.kotlin.util.PerformanceCounter.time(PerformanceCounter.kt:101)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.getTypeInfo(ExpressionTypingVisitorDispatcher.java:164)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.getTypeInfo(ExpressionTypingVisitorDispatcher.java:134)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorForStatements.visitExpression(ExpressionTypingVisitorForStatements.java:512)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorForStatements.visitExpression(ExpressionTypingVisitorForStatements.java:68)
	at org.jetbrains.kotlin.psi.KtVisitor.visitObjectLiteralExpression(KtVisitor.java:310)
	at org.jetbrains.kotlin.psi.KtObjectLiteralExpression.accept(KtObjectLiteralExpression.java:30)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.lambda$getTypeInfo$0(ExpressionTypingVisitorDispatcher.java:175)
	at org.jetbrains.kotlin.util.PerformanceCounter.time(PerformanceCounter.kt:101)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.getTypeInfo(ExpressionTypingVisitorDispatcher.java:164)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.getTypeInfo(ExpressionTypingVisitorDispatcher.java:147)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingServices.getTypeOfLastExpressionInBlock(ExpressionTypingServices.java:406)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingServices.getBlockReturnedTypeWithWritableScope(ExpressionTypingServices.java:327)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingServices.getBlockReturnedType(ExpressionTypingServices.java:206)
	at org.jetbrains.kotlin.types.expressions.FunctionsTypingVisitor.computeUnsafeReturnType(FunctionsTypingVisitor.kt:272)
	at org.jetbrains.kotlin.types.expressions.FunctionsTypingVisitor.computeReturnType(FunctionsTypingVisitor.kt:237)
	at org.jetbrains.kotlin.types.expressions.FunctionsTypingVisitor.visitLambdaExpression(FunctionsTypingVisitor.kt:174)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.visitLambdaExpression(ExpressionTypingVisitorDispatcher.java:262)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher$ForDeclarations.visitLambdaExpression(ExpressionTypingVisitorDispatcher.java:46)
	at org.jetbrains.kotlin.psi.KtLambdaExpression.accept(KtLambdaExpression.java:40)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.lambda$getTypeInfo$0(ExpressionTypingVisitorDispatcher.java:175)
	at org.jetbrains.kotlin.util.PerformanceCounter.time(PerformanceCounter.kt:101)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.getTypeInfo(ExpressionTypingVisitorDispatcher.java:164)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.getTypeInfo(ExpressionTypingVisitorDispatcher.java:134)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingServices.getTypeInfo(ExpressionTypingServices.java:125)
	at org.jetbrains.kotlin.resolve.calls.tower.KotlinResolutionCallbacksImpl.analyzeAndGetLambdaReturnArguments(KotlinResolutionCallbacksImpl.kt:190)
	at org.jetbrains.kotlin.resolve.calls.components.PostponedArgumentsAnalyzer.analyzeLambda(PostponedArgumentsAnalyzer.kt:128)
	at org.jetbrains.kotlin.resolve.calls.components.PostponedArgumentsAnalyzer.analyze(PostponedArgumentsAnalyzer.kt:41)
	at org.jetbrains.kotlin.resolve.calls.components.KotlinCallCompleter$runCompletion$1.invoke(KotlinCallCompleter.kt:229)
	at org.jetbrains.kotlin.resolve.calls.components.KotlinCallCompleter$runCompletion$1.invoke(KotlinCallCompleter.kt:219)
	at org.jetbrains.kotlin.resolve.calls.inference.components.KotlinConstraintSystemCompleter.analyzeArgumentWithFixedParameterTypes(KotlinConstraintSystemCompleter.kt:245)
	at org.jetbrains.kotlin.resolve.calls.inference.components.KotlinConstraintSystemCompleter.runCompletion(KotlinConstraintSystemCompleter.kt:81)
	at org.jetbrains.kotlin.resolve.calls.inference.components.KotlinConstraintSystemCompleter.runCompletion(KotlinConstraintSystemCompleter.kt:37)
	at org.jetbrains.kotlin.resolve.calls.components.KotlinCallCompleter.runCompletion(KotlinCallCompleter.kt:219)
	at org.jetbrains.kotlin.resolve.calls.components.KotlinCallCompleter.runCompletion$default(KotlinCallCompleter.kt:210)
	at org.jetbrains.kotlin.resolve.calls.components.KotlinCallCompleter.runCompletion(KotlinCallCompleter.kt:207)
	at org.jetbrains.kotlin.resolve.calls.components.KotlinCallCompleter.runCompletion(KotlinCallCompleter.kt:69)
	at org.jetbrains.kotlin.resolve.calls.KotlinCallResolver.choseMostSpecific(KotlinCallResolver.kt:162)
	at org.jetbrains.kotlin.resolve.calls.KotlinCallResolver.resolveCall(KotlinCallResolver.kt:82)
	at org.jetbrains.kotlin.resolve.calls.tower.PSICallResolver.runResolutionAndInference(PSICallResolver.kt:101)
	at org.jetbrains.kotlin.resolve.calls.CallResolver.doResolveCallOrGetCachedResults(CallResolver.java:601)
	at org.jetbrains.kotlin.resolve.calls.CallResolver.lambda$computeTasksAndResolveCall$0(CallResolver.java:213)
	at org.jetbrains.kotlin.util.PerformanceCounter.time(PerformanceCounter.kt:101)
	at org.jetbrains.kotlin.resolve.calls.CallResolver.computeTasksAndResolveCall(CallResolver.java:211)
	at org.jetbrains.kotlin.resolve.calls.CallResolver.computeTasksAndResolveCall(CallResolver.java:199)
	at org.jetbrains.kotlin.resolve.calls.CallResolver.resolveFunctionCall(CallResolver.java:329)
	at org.jetbrains.kotlin.resolve.calls.CallExpressionResolver.getResolvedCallForFunction(CallExpressionResolver.kt:88)
	at org.jetbrains.kotlin.resolve.calls.CallExpressionResolver.getCallExpressionTypeInfoWithoutFinalTypeCheck(CallExpressionResolver.kt:210)
	at org.jetbrains.kotlin.resolve.calls.CallExpressionResolver.getCallExpressionTypeInfo(CallExpressionResolver.kt:187)
	at org.jetbrains.kotlin.types.expressions.BasicExpressionTypingVisitor.visitCallExpression(BasicExpressionTypingVisitor.java:708)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.visitCallExpression(ExpressionTypingVisitorDispatcher.java:388)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher$ForDeclarations.visitCallExpression(ExpressionTypingVisitorDispatcher.java:46)
	at org.jetbrains.kotlin.psi.KtCallExpression.accept(KtCallExpression.java:35)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.lambda$getTypeInfo$0(ExpressionTypingVisitorDispatcher.java:175)
	at org.jetbrains.kotlin.util.PerformanceCounter.time(PerformanceCounter.kt:101)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.getTypeInfo(ExpressionTypingVisitorDispatcher.java:164)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.getTypeInfo(ExpressionTypingVisitorDispatcher.java:134)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.getTypeInfo(ExpressionTypingVisitorDispatcher.java:146)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingServices.getTypeInfo(ExpressionTypingServices.java:120)
	at org.jetbrains.kotlin.resolve.DelegatedPropertyResolver.resolveWithNewInference(DelegatedPropertyResolver.kt:573)
	at org.jetbrains.kotlin.resolve.DelegatedPropertyResolver.resolveDelegateExpression(DelegatedPropertyResolver.kt:506)
	at org.jetbrains.kotlin.resolve.VariableTypeAndInitializerResolver$resolveDelegatedPropertyType$1.invoke(VariableTypeAndInitializerResolver.kt:160)
	at org.jetbrains.kotlin.resolve.VariableTypeAndInitializerResolver$resolveDelegatedPropertyType$1.invoke(VariableTypeAndInitializerResolver.kt:158)
	at org.jetbrains.kotlin.storage.LockBasedStorageManager$LockBasedLazyValue.invoke(LockBasedStorageManager.java:408)
	at org.jetbrains.kotlin.storage.LockBasedStorageManager$LockBasedNotNullLazyValue.invoke(LockBasedStorageManager.java:527)
	at org.jetbrains.kotlin.types.DeferredType.getDelegate(DeferredType.java:106)
	at org.jetbrains.kotlin.resolve.BodyResolver.computeDeferredType(BodyResolver.java:1068)
	at org.jetbrains.kotlin.resolve.BodyResolver.resolveProperty(BodyResolver.java:788)
	at org.jetbrains.kotlin.resolve.BodyResolver.resolvePropertyDeclarationBodies(BodyResolver.java:846)
	at org.jetbrains.kotlin.resolve.BodyResolver.resolveBehaviorDeclarationBodies(BodyResolver.java:120)
	at org.jetbrains.kotlin.resolve.BodyResolver.resolveBodies(BodyResolver.java:253)
	at org.jetbrains.kotlin.resolve.LazyTopDownAnalyzer.analyzeDeclarations(LazyTopDownAnalyzer.kt:227)
	at org.jetbrains.kotlin.resolve.LazyTopDownAnalyzer.analyzeDeclarations$default(LazyTopDownAnalyzer.kt:58)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:127)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:96)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:262)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:53)
	at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:113)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyze(KotlinToJVMBytecodeCompiler.kt:253)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.repeatAnalysisIfNeeded(KotlinToJVMBytecodeCompiler.kt:223)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.repeatAnalysisIfNeeded(KotlinToJVMBytecodeCompiler.kt:223)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:100)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli$default(KotlinToJVMBytecodeCompiler.kt:58)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:170)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:52)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:92)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:44)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:98)
	at org.jetbrains.kotlin.incremental.IncrementalJvmCompilerRunner.runCompiler(IncrementalJvmCompilerRunner.kt:434)
	at org.jetbrains.kotlin.incremental.IncrementalJvmCompilerRunner.runCompiler(IncrementalJvmCompilerRunner.kt:120)
	at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.compileIncrementally(IncrementalCompilerRunner.kt:357)
	at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.compileIncrementally$default(IncrementalCompilerRunner.kt:299)
	at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.compileImpl(IncrementalCompilerRunner.kt:159)
	at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.compile(IncrementalCompilerRunner.kt:80)
	at org.jetbrains.kotlin.daemon.CompileServiceImplBase.execIncrementalCompiler(CompileServiceImpl.kt:622)
	at org.jetbrains.kotlin.daemon.CompileServiceImplBase.access$execIncrementalCompiler(CompileServiceImpl.kt:100)
	at org.jetbrains.kotlin.daemon.CompileServiceImpl.compile(CompileServiceImpl.kt:1713)
	at jdk.internal.reflect.GeneratedMethodAccessor105.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:360)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:200)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:197)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
	at java.rmi/sun.rmi.transport.Transport.serviceCall(Transport.java:196)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:587)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:828)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:705)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:704)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: kotlin.NotImplementedError: An operation is not implemented: Missing impl for by lazy { object : Any() {} } /home/wesley/work/projects/arrow-analysis-test/src/main/kotlin/ca/cutterslade/arrowanalysis/test/lazy-init.kt (org.jetbrains.kotlin.psi.KtPropertyDelegate)
	at arrow.meta.plugins.analysis.phases.analysis.solver.ast.kotlin.ast.KotlinInterpreterKt.model(KotlinInterpreter.kt:147)
	at arrow.meta.plugins.analysis.phases.analysis.solver.ast.kotlin.elements.KotlinDeclaration$DefaultImpls.getParents(KotlinDeclaration.kt:18)
	at arrow.meta.plugins.analysis.phases.analysis.solver.ast.kotlin.elements.KotlinNamedDeclaration$DefaultImpls.getParents(KotlinNamedDeclaration.kt:8)
	at arrow.meta.plugins.analysis.phases.analysis.solver.ast.kotlin.elements.KotlinClassOrObject$DefaultImpls.getParents(KotlinClassOrObject.kt:9)
	at arrow.meta.plugins.analysis.phases.analysis.solver.ast.kotlin.elements.KotlinObjectDeclaration.getParents(KotlinObjectDeclaration.kt:6)
	at arrow.meta.plugins.analysis.phases.analysis.solver.check.EntryPointKt.shouldBeAnalyzed(EntryPoint.kt:88)
	at arrow.meta.plugins.analysis.phases.analysis.solver.check.EntryPointKt.checkDeclarationConstraints(EntryPoint.kt:51)
	at arrow.meta.plugins.analysis.phases.PhasesKt$analysisPhases$4.invoke(Phases.kt:128)
	at arrow.meta.plugins.analysis.phases.PhasesKt$analysisPhases$4.invoke(Phases.kt:121)
	at arrow.meta.dsl.config.ConfigSyntax$storageComponent$1.check(ConfigSyntax.kt:72)
	at arrow.meta.internal.registry.InternalRegistry$DelegatingContributor$registerModuleComponents$2.check(InternalRegistry.kt:651)
	at org.jetbrains.kotlin.resolve.ModifiersChecker$ModifiersCheckingProcedure.runDeclarationCheckers(ModifiersChecker.java:285)
	at org.jetbrains.kotlin.resolve.ModifiersChecker$ModifiersCheckingProcedure.checkModifierListCommon(ModifiersChecker.java:224)
	at org.jetbrains.kotlin.resolve.ModifiersChecker$ModifiersCheckingProcedure.checkModifiersForDeclaration(ModifiersChecker.java:183)
	at org.jetbrains.kotlin.resolve.DeclarationsChecker.process(DeclarationsChecker.kt:94)
	at org.jetbrains.kotlin.resolve.BodyResolver.resolveBodies(BodyResolver.java:255)
	at org.jetbrains.kotlin.resolve.LazyTopDownAnalyzer.analyzeDeclarations(LazyTopDownAnalyzer.kt:227)
	at org.jetbrains.kotlin.types.expressions.LocalClassifierAnalyzer.processClassOrObject(LocalClassifierAnalyzer.kt:120)
	at org.jetbrains.kotlin.types.expressions.BasicExpressionTypingVisitor.visitObjectLiteralExpression(BasicExpressionTypingVisitor.java:672)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.visitObjectLiteralExpression(ExpressionTypingVisitorDispatcher.java:378)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher$ForBlock.visitObjectLiteralExpression(ExpressionTypingVisitorDispatcher.java:59)
	at org.jetbrains.kotlin.psi.KtObjectLiteralExpression.accept(KtObjectLiteralExpression.java:30)
	at org.jetbrains.kotlin.types.expressions.ExpressionTypingVisitorDispatcher.lambda$getTypeInfo$0(ExpressionTypingVisitorDispatcher.java:175)
	... 113 more
```

**Environment (please complete the following information):**
 - Kotlin version: `1.6.10`
 - Analysis plugin version: `io.arrow-kt.analysis.kotlin:io.arrow-kt.analysis.kotlin.gradle.plugin:2.0`

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,9 +1,9 @@
 # Package definitions
 projects.group=io.arrow-kt
-projects.version=1.6.0
-projects.meta_version=1.6.0
+projects.version=1.6.1-SNAPSHOT
+projects.meta_version=1.6.1-SNAPSHOT
 projects.optics_version=2.0-SNAPSHOT
-projects.analysis_version=2.0
+projects.analysis_version=2.0.1-SNAPSHOT
 projects.proofs_version=2.0-SNAPSHOT
 pom.description=Arrow Meta
 pom.url=https://github.com/arrow-kt/arrow-meta/

--- a/plugins/analysis/common/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/ast/context/types/Type.kt
+++ b/plugins/analysis/common/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/ast/context/types/Type.kt
@@ -30,4 +30,11 @@ interface Type : Comparable<Type> {
       other.isSubtypeOf(this) -> 1
       else -> 0
     }
+
+  fun intersectIfPossible(other: Type): Type? =
+    when {
+      other.isSubtypeOf(this) -> other
+      this.isSubtypeOf(other) -> this
+      else -> null
+    }
 }

--- a/plugins/analysis/common/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/check/Declarations.kt
+++ b/plugins/analysis/common/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/check/Declarations.kt
@@ -150,15 +150,15 @@ internal fun <A> SolverState.checkTopLevel(
 internal fun SolverState.checkTopLevelDeclarationWithBody(
   context: ResolutionContext,
   descriptor: DeclarationDescriptor,
-  declaration: Declaration
+  declaration: Declaration,
+  body: Expression?
 ): ContSeq<Unit> =
   checkTopLevel(context, descriptor, declaration, isConstructor = false, solver.resultVariable) {
     data,
     checkPost ->
     // only check body when we are not in a @Law
     doOnlyWhen(!descriptor.isALaw()) {
-      checkExpressionConstraints(solver.resultVariable, declaration.stableBody(), data).map {
-        finalState ->
+      checkExpressionConstraints(solver.resultVariable, body, data).map { finalState ->
         checkPost(finalState.data)
       }
     }

--- a/plugins/analysis/common/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/check/EntryPoint.kt
+++ b/plugins/analysis/common/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/check/EntryPoint.kt
@@ -2,14 +2,17 @@ package arrow.meta.plugins.analysis.phases.analysis.solver.check
 
 import arrow.meta.continuations.ContSeq
 import arrow.meta.continuations.doOnlyWhen
+import arrow.meta.continuations.doOnlyWhenNotNull
 import arrow.meta.plugins.analysis.phases.analysis.solver.ast.context.ResolutionContext
 import arrow.meta.plugins.analysis.phases.analysis.solver.ast.context.descriptors.ClassDescriptor
 import arrow.meta.plugins.analysis.phases.analysis.solver.ast.context.descriptors.DeclarationDescriptor
+import arrow.meta.plugins.analysis.phases.analysis.solver.ast.context.descriptors.PropertyDescriptor
 import arrow.meta.plugins.analysis.phases.analysis.solver.ast.context.elements.CallableDeclaration
 import arrow.meta.plugins.analysis.phases.analysis.solver.ast.context.elements.ClassOrObject
 import arrow.meta.plugins.analysis.phases.analysis.solver.ast.context.elements.Declaration
 import arrow.meta.plugins.analysis.phases.analysis.solver.ast.context.elements.EnumEntry
 import arrow.meta.plugins.analysis.phases.analysis.solver.ast.context.elements.PrimaryConstructor
+import arrow.meta.plugins.analysis.phases.analysis.solver.ast.context.elements.Property
 import arrow.meta.plugins.analysis.phases.analysis.solver.ast.context.elements.SecondaryConstructor
 import arrow.meta.plugins.analysis.phases.analysis.solver.ast.context.elements.TypeAlias
 import arrow.meta.plugins.analysis.phases.analysis.solver.errors.ErrorIds
@@ -65,7 +68,60 @@ public fun SolverState.checkDeclarationConstraints(
             descriptor as ClassDescriptor
             checkImplicitPrimaryConstructor(context, descriptor.constructors.single(), declaration)
           }
-        else -> checkTopLevelDeclarationWithBody(context, descriptor, declaration)
+        is Property ->
+          when {
+            declaration.stableBody() != null ->
+              checkTopLevelDeclarationWithBody(
+                context,
+                descriptor,
+                declaration,
+                declaration.stableBody()
+              )
+            declaration.delegate != null ->
+              checkTopLevelDeclarationWithBody(
+                context,
+                descriptor,
+                declaration,
+                declaration.delegate?.expression
+              )
+            declaration.getter != null || declaration.setter != null -> {
+              // no body, maybe we have getter and setter
+              descriptor as PropertyDescriptor
+              ContSeq.unit
+                .map {
+                  doOnlyWhenNotNull(declaration.getter, Unit) { getterDecl ->
+                    doOnlyWhenNotNull(descriptor.getter, Unit) { getterDescr ->
+                      checkTopLevelDeclarationWithBody(
+                        context,
+                        getterDescr,
+                        getterDecl,
+                        getterDecl.body()
+                      )
+                    }
+                  }
+                }
+                .map {
+                  doOnlyWhenNotNull(declaration.setter, Unit) { setterDecl ->
+                    doOnlyWhenNotNull(descriptor.setter, Unit) { setterDescr ->
+                      checkTopLevelDeclarationWithBody(
+                        context,
+                        setterDescr,
+                        setterDecl,
+                        setterDecl.body()
+                      )
+                    }
+                  }
+                }
+            }
+            else -> ContSeq.unit
+          }
+        else ->
+          checkTopLevelDeclarationWithBody(
+            context,
+            descriptor,
+            declaration,
+            declaration.stableBody()
+          )
       }.drain()
     } catch (e: IllegalStateException) {
       val msg = ErrorMessages.Exception.illegalState(solverTrace)

--- a/plugins/analysis/common/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/collect/FromDSL.kt
+++ b/plugins/analysis/common/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/collect/FromDSL.kt
@@ -259,8 +259,13 @@ private fun Element.elementToConstraint(
     val result = solverState.topLevelExpressionToFormula(predicateArg, context, parameters, false)
     if (result == null) {
       val msg = ErrorMessages.Parsing.errorParsingPredicate(predicateArg)
-      context.handleError(ErrorIds.Parsing.ErrorParsingPredicate, this, msg)
-      solverState.signalParseErrors()
+      if (call.isRequireCall() || call.isAssertCall()) {
+        // for built-ins only issue a warning
+        context.handleError(ErrorIds.Parsing.WarningParsingPredicate, this, msg)
+      } else {
+        context.handleError(ErrorIds.Parsing.ErrorParsingPredicate, this, msg)
+        solverState.signalParseErrors()
+      }
       null
     } else {
       val msgBody = call.arg("msg", context) ?: call.arg("lazyMessage", context)

--- a/plugins/analysis/common/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/errors/ErrorIds.kt
+++ b/plugins/analysis/common/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/errors/ErrorIds.kt
@@ -20,6 +20,7 @@ sealed interface ErrorIds {
 
   enum class Parsing : ErrorIds {
     ErrorParsingPredicate,
+    WarningParsingPredicate,
     UnexpectedReference,
     UnexpectedFieldInitBlock;
 

--- a/plugins/analysis/common/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/search/ConstraintSearch.kt
+++ b/plugins/analysis/common/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/search/ConstraintSearch.kt
@@ -187,15 +187,32 @@ internal fun SolverState.primitiveConstraints(
   val returnTy =
     if (descriptor.isComparison()) PrimitiveType.BOOLEAN
     else descriptor.returnType?.unwrapIfNullable()?.primitiveType()
+  // inference doesn't give us the correct type from call sometimes,
+  // so look also at the given expression and intersect the types
   val dispatch =
     listOfNotNull(descriptor.extensionReceiverParameter, descriptor.dispatchReceiverParameter)
       .map { param ->
-        param.type.unwrapIfNullable().primitiveType()?.let { ty -> Pair(THIS_VAR_NAME, ty) }
+        val tyFromArg = call.getReceiverExpression()?.type(context)
+        val tyFromParam = param.type.unwrapIfNullable()
+        val chosenTy = tyFromArg?.intersectIfPossible(tyFromParam) ?: tyFromParam
+        chosenTy.primitiveType()?.let { ty -> Pair(THIS_VAR_NAME, ty) }
       }
   val argTys =
     dispatch +
       descriptor.valueParameters.map { param ->
-        param.type.unwrapIfNullable().primitiveType()?.let { ty -> Pair(param.name.value, ty) }
+        val tyFromArg =
+          call
+            .valueArguments
+            .filterKeys { it.name == param.name }
+            .values
+            .singleOrNull()
+            ?.arguments
+            ?.singleOrNull()
+            ?.argumentExpression
+            ?.type(context)
+        val tyFromParam = param.type.unwrapIfNullable()
+        val chosenTy = tyFromArg?.intersectIfPossible(tyFromParam) ?: tyFromParam
+        chosenTy.primitiveType()?.let { ty -> Pair(param.name.value, ty) }
       }
   return if (returnTy == null || argTys.any { it == null }) {
     null

--- a/plugins/analysis/java-plugin/src/main/kotlin/arrow/meta/plugins/analysis/java/AnalysisMessages.kt
+++ b/plugins/analysis/java-plugin/src/main/kotlin/arrow/meta/plugins/analysis/java/AnalysisMessages.kt
@@ -17,6 +17,7 @@ public class AnalysisMessages : ListResourceBundle() {
     public const val UnsatInvariants: String = "unsat_invariants"
     public const val LiskovProblem: String = "liskov_problem"
     public const val ErrorParsingPredicate: String = "error_parsing_predicate"
+    public const val WarningParsingPredicate: String = "warning_parsing_predicate"
     public const val UnsupportedElement: String = "unsupported_element"
     public const val AnalysisException: String = "analysis_exception"
 
@@ -33,7 +34,12 @@ public class AnalysisMessages : ListResourceBundle() {
       )
 
     public val Warnings: List<String> =
-      listOf(InconsistentCallPost, InconsistentConditions, UnsupportedElement)
+      listOf(
+        InconsistentCallPost,
+        InconsistentConditions,
+        UnsupportedElement,
+        WarningParsingPredicate
+      )
   }
 
   // "err" and "warn" come from [DiagnosticType.key]

--- a/plugins/analysis/java-plugin/src/main/kotlin/arrow/meta/plugins/analysis/java/ast/JavaResolutionContext.kt
+++ b/plugins/analysis/java-plugin/src/main/kotlin/arrow/meta/plugins/analysis/java/ast/JavaResolutionContext.kt
@@ -89,6 +89,7 @@ public class JavaResolutionContext(
       ErrorIds.Liskov.NotWeakerPrecondition -> reportLiskovProblem(element, msg)
       ErrorIds.Liskov.NotStrongerPostcondition -> reportLiskovProblem(element, msg)
       ErrorIds.Parsing.ErrorParsingPredicate -> reportErrorsParsingPredicate(element, msg)
+      ErrorIds.Parsing.WarningParsingPredicate -> reportWarningsParsingPredicate(element, msg)
       ErrorIds.Parsing.UnexpectedReference -> reportErrorsParsingPredicate(element, msg)
       ErrorIds.Parsing.UnexpectedFieldInitBlock -> reportErrorsParsingPredicate(element, msg)
       ErrorIds.Laws.LawMustCallFunction -> reportErrorsParsingPredicate(element, msg)
@@ -140,6 +141,9 @@ public class JavaResolutionContext(
 
   private fun reportErrorsParsingPredicate(element: Element, msg: String): Unit =
     reportError(element, AnalysisMessages.ErrorParsingPredicate, msg)
+
+  private fun reportWarningsParsingPredicate(element: Element, msg: String): Unit =
+    reportWarning(element, AnalysisMessages.WarningParsingPredicate, msg)
 
   private fun reportUnsatCallPre(element: Element, msg: String): Unit =
     reportError(element, AnalysisMessages.UnsatCallPre, msg)

--- a/plugins/analysis/kotlin-plugin/src/main/java/arrow/meta/plugins/analysis/errors/MetaDefaultErrorMessagesJvm.java
+++ b/plugins/analysis/kotlin-plugin/src/main/java/arrow/meta/plugins/analysis/errors/MetaDefaultErrorMessagesJvm.java
@@ -28,6 +28,7 @@ public class MetaDefaultErrorMessagesJvm implements DefaultErrorMessages.Extensi
         MAP.put(UnsatInvariants, "{0}", RenderString.instance);
         MAP.put(LiskovProblem, "{0}", RenderString.instance);
         MAP.put(ErrorParsingPredicate, "{0}", RenderString.instance);
+        MAP.put(WarningParsingPredicate, "{0}", RenderString.instance);
         MAP.put(UnsupportedElement, "{0}", RenderString.instance);
         MAP.put(AnalysisException, "{0}", RenderString.instance);
     }

--- a/plugins/analysis/kotlin-plugin/src/main/java/arrow/meta/plugins/analysis/errors/MetaErrors.java
+++ b/plugins/analysis/kotlin-plugin/src/main/java/arrow/meta/plugins/analysis/errors/MetaErrors.java
@@ -18,6 +18,7 @@ public interface MetaErrors {
     DiagnosticFactory1<PsiElement, String> UnsatInvariants = DiagnosticFactory1.create(ERROR);
     DiagnosticFactory1<PsiElement, String> LiskovProblem = DiagnosticFactory1.create(ERROR);
     DiagnosticFactory1<PsiElement, String> ErrorParsingPredicate = DiagnosticFactory1.create(ERROR);
+    DiagnosticFactory1<PsiElement, String> WarningParsingPredicate = DiagnosticFactory1.create(WARNING);
     DiagnosticFactory1<PsiElement, String> UnsupportedElement = DiagnosticFactory1.create(WARNING);
     DiagnosticFactory1<PsiElement, String> AnalysisException = DiagnosticFactory1.create(ERROR);
 

--- a/plugins/analysis/kotlin-plugin/src/main/kotlin/arrow/meta/plugins/analysis/phases/Phases.kt
+++ b/plugins/analysis/kotlin-plugin/src/main/kotlin/arrow/meta/plugins/analysis/phases/Phases.kt
@@ -22,6 +22,7 @@ import arrow.meta.plugins.analysis.phases.ir.HintState
 import arrow.meta.plugins.analysis.phases.ir.annotateWithConstraints
 import arrow.meta.plugins.analysis.phases.ir.hintsFile
 import kotlin.io.path.Path
+import org.jetbrains.kotlin.diagnostics.Diagnostic
 import org.jetbrains.kotlin.psi.KtDeclaration
 import org.jetbrains.kotlin.psi.KtFile
 import org.jetbrains.kotlin.resolve.checkers.DeclarationCheckerContext
@@ -56,6 +57,8 @@ internal fun Meta.analysisPhases(): ExtensionPhase =
               val locals = files.declarationDescriptors(context)
               val (result, interesting) =
                 solverState.collectConstraintsFromAnnotations(locals, kotlinModule, context)
+              val currentParseWarnings = bindingTrace.bindingContext.diagnostics.toList()
+              setParseWarningsAs(module, currentParseWarnings)
               setStageAs(module, Stage.Prove) // we end the CollectConstraints phase
               when (result) {
                 AnalysisResult.Retry -> {
@@ -98,6 +101,11 @@ internal fun Meta.analysisPhases(): ExtensionPhase =
               null
             }
           } else {
+            // HACK: because diagnostics are cleared after retrying
+            //       we keep them in a temporary place and re-issue them
+            if (isInStage(module, Stage.Prove)) {
+              getParseWarnings(module)?.forEach { bindingTrace.report(it) }
+            }
             val solverState = solverState(module)
             solverState?.notifyModuleProcessed(KotlinModuleDescriptor(module))
             null
@@ -155,6 +163,12 @@ object Keys {
     "SolverState-${moduleDescriptor.name}"
 
   fun hints(module: ModuleDescriptor): String = "arrow-analysis-hint-${module.name}"
+
+  fun parseWarnings(module: ModuleDescriptor): String =
+    "arrow-analysis-parse-warnings-${module.name}"
+
+  fun parseWarnings(module: org.jetbrains.kotlin.descriptors.ModuleDescriptor): String =
+    "arrow-analysis-parse-warnings-${module.name}"
 }
 
 internal fun CompilerContext.solverState(context: DeclarationCheckerContext): SolverState? =
@@ -179,6 +193,7 @@ internal fun CompilerContext.initialize(module: ModuleDescriptor) {
   }
   ensureInitialized(Keys.hints(module)) { HintState.NeedsProcessing }
   ensureInitialized(Keys.stage(module)) { Stage.Init }
+  ensureInitialized(Keys.parseWarnings(module)) { emptyList<Diagnostic>() }
 }
 
 internal inline fun <reified A : Any> CompilerContext.ensureInitialized(
@@ -213,3 +228,14 @@ private fun CompilerContext.setStageAs(
 ) {
   set(Keys.stage(module), stage)
 }
+
+private fun CompilerContext.setParseWarningsAs(
+  module: org.jetbrains.kotlin.descriptors.ModuleDescriptor,
+  warnings: List<Diagnostic>
+) {
+  set(Keys.parseWarnings(module), warnings)
+}
+
+private fun CompilerContext.getParseWarnings(
+  module: org.jetbrains.kotlin.descriptors.ModuleDescriptor
+) = get<List<Diagnostic>>(Keys.parseWarnings(module))

--- a/plugins/analysis/kotlin-plugin/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/ast/kotlin/KotlinResolutionContext.kt
+++ b/plugins/analysis/kotlin-plugin/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/ast/kotlin/KotlinResolutionContext.kt
@@ -112,6 +112,7 @@ class KotlinResolutionContext(
       ErrorIds.Liskov.NotWeakerPrecondition -> reportLiskovProblem(element, msg)
       ErrorIds.Liskov.NotStrongerPostcondition -> reportLiskovProblem(element, msg)
       ErrorIds.Parsing.ErrorParsingPredicate -> reportErrorsParsingPredicate(element, msg)
+      ErrorIds.Parsing.WarningParsingPredicate -> reportWarningsParsingPredicate(element, msg)
       ErrorIds.Parsing.UnexpectedReference -> reportErrorsParsingPredicate(element, msg)
       ErrorIds.Parsing.UnexpectedFieldInitBlock -> reportErrorsParsingPredicate(element, msg)
       ErrorIds.Laws.LawMustCallFunction -> reportErrorsParsingPredicate(element, msg)
@@ -128,6 +129,10 @@ class KotlinResolutionContext(
 
   private fun reportErrorsParsingPredicate(element: Element, msg: String) {
     report(MetaErrors.ErrorParsingPredicate.on(element.element(), msg))
+  }
+
+  private fun reportWarningsParsingPredicate(element: Element, msg: String) {
+    report(MetaErrors.WarningParsingPredicate.on(element.element(), msg))
   }
 
   private fun reportUnsatCallPre(element: Element, msg: String) {

--- a/plugins/analysis/kotlin-plugin/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/ast/kotlin/ast/KotlinInterpreter.kt
+++ b/plugins/analysis/kotlin-plugin/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/ast/kotlin/ast/KotlinInterpreter.kt
@@ -105,6 +105,7 @@ fun <A : KtElement, B : Element> A.model(): B =
     is KtParameterList -> KotlinParameterList(this).repr()
     is KtIsExpression -> KotlinIsExpression(this).repr()
     is KtTypeReference -> KotlinTypeReference(this).repr()
+    is KtThrowExpression -> KotlinThrowExpression(this).repr()
     is KtTryExpression -> KotlinTryExpression(this).repr()
     is KtCatchClause -> KotlinCatchClause(this).repr()
     is KtFinallySection -> KotlinFinallySection(this).repr()

--- a/plugins/analysis/kotlin-plugin/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/ast/kotlin/ast/KotlinInterpreter.kt
+++ b/plugins/analysis/kotlin-plugin/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/ast/kotlin/ast/KotlinInterpreter.kt
@@ -79,6 +79,7 @@ fun <A : KtElement, B : Element> A.model(): B =
     is KtClass -> KotlinClass(this).repr()
     is KtObjectDeclaration -> KotlinObjectDeclaration(this).repr()
     is KtObjectLiteralExpression -> KotlinObjectLiteralExpression(this).repr()
+    is KtPropertyDelegate -> KotlinPropertyDelegate(this).repr()
     is KtClassBody -> KotlinClassBody(this).repr()
     is KtLambdaExpression -> KotlinLambdaExpression(this).repr()
     is KtValueArgument -> KotlinValueArgument(this).repr()
@@ -141,6 +142,8 @@ fun <A : KtElement, B : Element> A.model(): B =
     is KtContainerNodeForControlStructureBody -> this.expression?.model()
         ?: KotlinDefaultElement(this).repr()
     is KtContainerNode -> KotlinDefaultElement(this).repr()
+    // final fallback for expressions
+    is KtExpression -> KotlinDefaultExpression(this).repr()
     // is KtFile -> KotlinFile(this).repr()
     else ->
       TODO(

--- a/plugins/analysis/kotlin-plugin/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/ast/kotlin/elements/KotlinPropertyDelegate.kt
+++ b/plugins/analysis/kotlin-plugin/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/ast/kotlin/elements/KotlinPropertyDelegate.kt
@@ -5,8 +5,8 @@ import arrow.meta.plugins.analysis.phases.analysis.solver.ast.context.elements.P
 import arrow.meta.plugins.analysis.phases.analysis.solver.ast.kotlin.ast.model
 import org.jetbrains.kotlin.psi.KtPropertyDelegate
 
-fun interface KotlinPropertyDelegate : PropertyDelegate, KotlinElement {
-  override fun impl(): KtPropertyDelegate
+class KotlinPropertyDelegate(val impl: KtPropertyDelegate) : PropertyDelegate, KotlinElement {
+  override fun impl(): KtPropertyDelegate = impl
   override val expression: Expression?
     get() = impl().expression?.model()
 }

--- a/plugins/analysis/kotlin-plugin/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/ast/kotlin/elements/KotlinThrowExpression.kt
+++ b/plugins/analysis/kotlin-plugin/src/main/kotlin/arrow/meta/plugins/analysis/phases/analysis/solver/ast/kotlin/elements/KotlinThrowExpression.kt
@@ -5,8 +5,8 @@ import arrow.meta.plugins.analysis.phases.analysis.solver.ast.context.elements.T
 import arrow.meta.plugins.analysis.phases.analysis.solver.ast.kotlin.ast.model
 import org.jetbrains.kotlin.psi.KtThrowExpression
 
-fun interface KotlinThrowExpression : ThrowExpression, KotlinExpression {
-  override fun impl(): KtThrowExpression
+class KotlinThrowExpression(val impl: KtThrowExpression) : ThrowExpression, KotlinExpression {
+  override fun impl(): KtThrowExpression = impl
   override val thrownExpression: Expression?
     get() = impl().thrownExpression?.model()
 }

--- a/plugins/analysis/kotlin-plugin/src/test/kotlin/arrow/meta/plugins/analysis/AnalysisTests.kt
+++ b/plugins/analysis/kotlin-plugin/src/test/kotlin/arrow/meta/plugins/analysis/AnalysisTests.kt
@@ -1809,6 +1809,21 @@ class AnalysisTests {
       withoutPlugin = { compiles }
     )
   }
+
+  @Test
+  fun `throw does not break`() {
+    """
+      ${imports()}
+      
+      fun f(n: Int) = when {
+        n > 0 -> n + 1
+        else -> throw IllegalArgumentException("positive")
+      }
+      """(
+      withPlugin = { compilesNoUnreachable },
+      withoutPlugin = { compiles }
+    )
+  }
 }
 
 private val AssertSyntax.compilesNoUnreachable: Assert.SingleAssert

--- a/plugins/analysis/kotlin-plugin/src/test/kotlin/arrow/meta/plugins/analysis/AnalysisTests.kt
+++ b/plugins/analysis/kotlin-plugin/src/test/kotlin/arrow/meta/plugins/analysis/AnalysisTests.kt
@@ -53,6 +53,20 @@ class AnalysisTests {
   }
 
   @Test
+  fun `bad predicate, in require`() {
+    """
+      ${imports()}
+      fun bar(): Int {
+        require( "a" == "b" ) { "wrong" }
+        return 1
+      }
+      """(
+      withPlugin = { compilesWith { it.contains("not parse predicate") } },
+      withoutPlugin = { compiles }
+    )
+  }
+
+  @Test
   fun `inconsistent preconditions`() {
     """
       ${imports()}

--- a/plugins/analysis/kotlin-plugin/src/test/kotlin/arrow/meta/plugins/analysis/AnalysisTests.kt
+++ b/plugins/analysis/kotlin-plugin/src/test/kotlin/arrow/meta/plugins/analysis/AnalysisTests.kt
@@ -1838,6 +1838,25 @@ class AnalysisTests {
       withoutPlugin = { compiles }
     )
   }
+
+  @Test
+  fun `issue #1002`() {
+    """
+      ${imports()}
+      ${collectionListLaws()}
+      
+      fun domain(value: String): String? {
+        val parts = value.split("@") // no pattern matching in kotlin :((((
+        return if (parts.size != 2)
+            null
+        else
+            parts[1]
+      }
+      """(
+      withPlugin = { compilesNoUnreachable },
+      withoutPlugin = { compiles }
+    )
+  }
 }
 
 private val AssertSyntax.compilesNoUnreachable: Assert.SingleAssert
@@ -1894,6 +1913,11 @@ object ListLaws {
   @Law
   inline fun <E> List<E>.isEmptyLaw(): Boolean =
     isEmpty().post({ it == (size <= 0) }) { "empty list has size 0" }
+  @Law
+  inline fun <E> List<E>.getLaw(index: Int): E {
+    pre(index >= 0 && index < size) { "index within bounds" }
+    return get(index)
+  }
       
 }
 

--- a/plugins/analysis/kotlin-plugin/src/test/kotlin/arrow/meta/plugins/analysis/AnalysisTests.kt
+++ b/plugins/analysis/kotlin-plugin/src/test/kotlin/arrow/meta/plugins/analysis/AnalysisTests.kt
@@ -1784,6 +1784,31 @@ class AnalysisTests {
       withoutPlugin = { compiles }
     )
   }
+
+  @Test
+  fun `delegated property, ok`() {
+    """
+      ${imports()}
+      
+      val p by lazy { object : Any() {} }
+      """(
+      withPlugin = { compilesNoUnreachable },
+      withoutPlugin = { compiles }
+    )
+  }
+
+  @Test
+  fun `delegated property, wrong`() {
+    """
+      ${imports()}
+      ${collectionListLaws()}
+      
+      val p: Int by lazy { emptyList<Int>().first() }
+      """(
+      withPlugin = { failsWith { it.contains("pre-condition `not empty` is not satisfied") } },
+      withoutPlugin = { compiles }
+    )
+  }
 }
 
 private val AssertSyntax.compilesNoUnreachable: Assert.SingleAssert

